pkg/apprepo/apprepos_handler.go

```diff
@@ -151,7 +153,9 @@ func clientsetForConfig(config *rest.Config) (combinedClientsetInterface, error)
 // ConfigForToken returns a new config for a given auth token.
 func (a *AppRepositoriesHandler) ConfigForToken(token string) *rest.Config {
 	configCopy := a.config
-	configCopy.BearerToken = token
+	if token != "" {
+		configCopy.BearerToken = token
+	}
```

Comment:
This looks dangerous - as though it implicitly returns a service account config if you pass in an empty token? `ConfigForToken` should only ever return a config for the specified token, whether or not the token is passed with bad information.